pkg/factory/risingwave_object_factory.go

```diff
@@ -362,6 +362,12 @@ func (f *RisingWaveObjectFactory) NewCompactorService() *corev1.Service {
 // NewConnectorService creates a new Service for the connector.
 func (f *RisingWaveObjectFactory) NewConnectorService() *corev1.Service {
 	connectorPorts := &f.risingwave.Spec.Components.Connector.Ports
+	if connectorPorts.ServicePort == 0 {
```

Comment:
It would be better if you can make it a function and invoke it like

```go
connectorPorts := getConnectorPorts(f.risingwave)
// or
connectorPorts := f.getConnectorPorts()
```